src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -440,6 +440,15 @@ internal static bool IsOsUnixLike()
             return NativeMethodsShared.IsUnixLike;
         }
 
+        /// <summary>
+        /// True if current OS is a BSD system.
+        /// </summary>
+        /// <returns></returns>
+        internal static bool IsOsBsdLike()
```

Comment:
@Microsoft/msbuild-maintainers 

Are you OK with this new API? I added a `Is*Like` method because in our NativeMethodsShared code detecting bsd systems is more complicated than just calling into IsOsPlatform: https://github.com/Microsoft/msbuild/blob/master/src/Shared/NativeMethodsShared.cs#L496-L508

Reply:
Why do you think it's needed? What decisions need to be based on the super-granular nature of this?

Reply:
We at least need to disambiguate for BSD: https://github.com/Microsoft/msbuild/pull/2906/files/66d6c2e61df876ab49b110d6318a5f98140a32c9#diff-4b2f402501d23abbede6eac0f47783e4R48

Reply:
FYI @wfurt